### PR TITLE
Added criteria for backend takehome as well

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Adds a shared `interviews/README.md` with the common FAQ-style explanations between backend & frontend takehome grading. Copies over the rubric we've been using for backend grading.

As discussed with @hoffm, we'll likely have a few action items after this intern hiring season:

* Look into consolidating the two rubrics to be more similar
* Create a separate "takehome-grading" repo or similar
* Switch our `internship-code-challenge` repo to be closed source when not in interview season _(heh)_

### CODE REVIEW

File: interviews/backend-takehome.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+Data structures are chosen appropriately for low complexity and low memory allocation
+
+- 💖 4 - Appropriate data structures; efficient use of conditionals/loops; minimal array scanning; appropriate passing of parameters and return values
+- 👍 3 - Reasonable use of data structures; some opportunity for reduction of costly repeated operations; some inefficiency in usage of parameters and return values
+- 👎 2 - Excessive hardcoding of control flow and repetition of the same logic
+- 🛑 1 - Can enter unintended infinite loops; may not function properly; shows lack of understanding of data structures
+
+### Testing
+
+While unit tests are not required, some form of test that verifies that the code is working appropriately needs to exist
```
Comment (rogervera) [inline interviews/backend-takehome.md]: This doesn't correlate with wording in the backend prompt: [Unit tests are written to determine that code is correct.](https://github.com/Codecademy/internship-code-challenge/tree/master/backend#how-you-will-be-evaluated)

And Michael's [comment here](https://codecademy.slack.com/archives/GQWH1EPJ7/p1580152302013600) leads me to believe we haven't led to a consensus yet.

I'm for `unit tests or script`, but we can have that discussion in the appropriate channel.

Comment (JoshuaKGoldberg) [reply interviews/backend-takehome.md]: Aha, excellent callout! I'll adjust here...

File: interviews/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Takehome Grading
+
+Hi!
+This is our grading rubric for takehome projects.
+Thanks for taking a look!
+
+These are the scores by which our engineering team evaluate backend takehome assignments.
```
Comment (hoffm) [inline interviews/README.md]:
```suggestion
These are the criteria our engineering team uses to evaluate takehome assignments.
```

File: interviews/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+We _don't_ establish a hard minimum of score for any opening -- these scores are a general rubric, not a hard rule.
```
Comment (hoffm) [inline interviews/README.md]:
```suggestion
We don't set a minimum score for any opening, and although we take these criteria very seriously, we also reserve the right to use considerations beyond these rubrics in our decision making.
```

File: interviews/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Takehome Grading
+
+Hi!
+This is our grading rubric for takehome projects.
```
Comment (hoffm) [inline interviews/README.md]:
```suggestion
These are our grading guidelines for takehome projects.
```

File: interviews/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+Thanks for taking a look!
+
+These are the scores by which our engineering team evaluate backend takehome assignments.
+We _don't_ establish a hard minimum of score for any opening -- these scores are a general rubric, not a hard rule.
+
+## Rubrics
+
```
Comment (hoffm) [inline interviews/README.md]: Insert:

> Here are the specific grading rubrics we use to evaluate each of our takehome assignments:

File: interviews/README.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+- [Intern Backend Takehome](./frontend-takehome.md)
+- [Intern Frontend Takehome](./backend-takehome.md)
+
+## Why?
+
+We believe in openness and transparency in our engineering practices.
+There must be _some_ criteria by which we evaluate takehome submissions, and if candidates are meant to be taking them, we'd like them to be clear to everyone.
```
Comment (hoffm) [inline interviews/README.md]: I don't understand what "if candidates are meant to be taking them" means.

File: interviews/README.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+We also welcome any feedback you have on these!
+
+## Takehome Assignment Specifications
```
Comment (hoffm) [inline interviews/README.md]: I'm not sure whether this section belongs here. This section seems to be about what makes a good takehome assignment, whereas the rest of this document describes how we evaluate takehome submissions.

File: interviews/backend-takehome.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
+# Backend Takehome Grading
+
+## Section Scores
+
+### Solution Completion
+
+All requirements are met as outlined by the assignment's requirements.
+
+- 💖 4 - Perfect
```
Comment (hoffm) [inline interviews/backend-takehome.md]:
```suggestion
- 💖 4 - All stated requirements satisfied.
```

File: interviews/backend-takehome.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+- 👍 3 - One minor bug with flow using expected inputs, or a couple bugs with validation of corner cases
```
Comment (hoffm) [inline interviews/backend-takehome.md]:
```suggestion
- 👍 3 - One minor bug with flow using expected inputs or a couple of bugs around corner cases
```

File: interviews/backend-takehome.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+- 👎 2 - Two or more missed requirements
+- 🛑 1 - Several requirements missed; compile errors that are not trivial to resolve
+
+### Code Quality
+
+Code is generally well-organized and easy to understand.
+
+- 💖 4 - Well-organized, DRY, easily readable, good variable/function names, would pass linter w/minimal issues
```
Comment (hoffm) [inline interviews/backend-takehome.md]:
```suggestion
- 💖 4 - Well-organized, DRY, easily readable, good variable/function names, would pass linter with few issues
```

File: interviews/backend-takehome.md [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+- 👍 3 - Minor repetition in logic, a couple confusing variable/function names, minor formatting issues
```
Comment (hoffm) [inline interviews/backend-takehome.md]:
```suggestion
- 👍 3 - Minor repetition in logic, a couple of confusing variable/function names, minor formatting issues
```

File: interviews/backend-takehome.md [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+- 👎 2 - Not idiomatic for language chosen, difficult to understand and trace
+- 🛑 1 - Unable to understand, non-standard formatting, variable names difficult to follow
```
Comment (hoffm) [inline interviews/backend-takehome.md]:
```suggestion
- 🛑 1 - Very difficult to understand, non-standard formatting, variable names confusing
```

File: interviews/backend-takehome.md [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+
+### Data structures and runtime
+
+Data structures are chosen appropriately for low complexity and low memory allocation.
+
+- 💖 4 - Appropriate data structures; efficient use of conditionals/loops; minimal array scanning; appropriate passing of parameters and return values
```
Comment (hoffm) [inline interviews/backend-takehome.md]: Let's choose either commas or semicolons and stick to it.

File: interviews/backend-takehome.md [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+Code is generally well-organized and easy to understand.
+
+- 💖 4 - Well-organized, DRY, easily readable, good variable/function names, would pass linter w/minimal issues
+- 👍 3 - Minor repetition in logic, a couple confusing variable/function names, minor formatting issues
+- 👎 2 - Not idiomatic for language chosen, difficult to understand and trace
+- 🛑 1 - Unable to understand, non-standard formatting, variable names difficult to follow
+
+### Data structures and runtime
+
+Data structures are chosen appropriately for low complexity and low memory allocation.
```
Comment (hoffm) [inline interviews/backend-takehome.md]: The list below seems to be about code organization as well as data structures (e.g. "repetition of the same logic"). Maybe we want a more general description here.

File: interviews/backend-takehome.md [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+- 💖 4 - Appropriate data structures; efficient use of conditionals/loops; minimal array scanning; appropriate passing of parameters and return values
+- 👍 3 - Reasonable use of data structures; some opportunity for reduction of costly repeated operations; some inefficiency in usage of parameters and return values
+- 👎 2 - Excessive hardcoding of control flow and repetition of the same logic
+- 🛑 1 - Can enter unintended infinite loops; may not function properly; shows lack of understanding of data structures
+
+### Testing
+
+Unit tests verify the code is working appropriately.
+
+- 💖 4 - Unit tests or script that tests various logic branches including edge cases; tests are granular and tied to specific functions
```
Comment (hoffm) [inline interviews/backend-takehome.md]:
```suggestion
- 💖 4 - Unit tests or script that tests all non-trivial logic branches, including edge cases; tests are granular and tied to specific functions
```

File: interviews/backend-takehome.md [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+- 👍 3 - Unit tests or script that tests main logic branches but not edge cases; tests not tied to specific functions but rather groups of them
+- 👎 2 - Unit tests/script are not repeatable
+- 🛑 1 - No unit tests or scripts
+
+### Bonus
+
+This category could be used as a tie-breaker for candidates that scored similarly in other categories. Please [review the challenge requirements for the bonus](https://github.com/Codecademy/internship-code-challenge/tree/master/backend#bonus)
```
Comment (hoffm) [inline interviews/backend-takehome.md]:
```suggestion
This category may be used as a tie-breaker for candidates who score similarly in other categories. Please [review the challenge requirements for the bonus](https://github.com/Codecademy/internship-code-challenge/tree/master/backend#bonus)
```

File: interviews/README.md (hunk 7 above)
Comment (JoshuaKGoldberg) [reply interviews/README.md]: 🤔 interesting. Filed a followup issue to discuss: #18
